src/ngraph/runtime/backend.cpp

```diff
@@ -62,16 +51,24 @@ void* runtime::Backend::open_shared_library(string type)
     {
         type = type.substr(0, colon);
     }
-    string lib_name = "lib" + to_lower(type) + "_backend" + ext;
+
+    vector<string> backend_list;
+    string version = NGRAPH_VERSION;
+    version = regex_replace(version, regex("\\."), "\\.");
+    version = regex_replace(version, regex("\\+"), "\\+");
+    regex reg("lib" + to_lower(type) + "_backend.*" + version);
+    smatch result;
+
+    auto f = [&](const string& file, bool is_dir) {
+        string name = file_util::get_file_name(file);
+        if (regex_match(name, result, reg))
+        {
+            handle = dlopen(file.c_str(), RTLD_NOW | RTLD_GLOBAL);
```

Comment:
Would be good to keep the `runtime_error` here if `dlopen` fails, and perhaps throw rather than returning `nullptr` if the .so is not found.